cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -112,20 +112,11 @@ protected int getInstances(HttpRequest request) throws BadRequestException {
   }
 
   protected Map<String, String> decodeArguments(HttpRequest request) throws JsonSyntaxException {
-    ChannelBuffer content = request.getContent();
-    if (!content.readable()) {
+    Map<String, String> args = parseBody(request, STRING_MAP_TYPE);
```

Comment:
yes, why rely on a try...catch in client code? Unless you think that's beyond what ``parseBody``'s contract should be. But if we're going to have this exact same try block everywhere, it perhaps make sense to throw a ``BadRequestException`` from here itself. ``parseBody`` will never be used outside of HTTP Handler code right?